lib/src/builder/codegen/typed_map_impl_generator.dart

```diff
@@ -390,11 +390,15 @@ class _TypedMapImplGenerator extends TypedMapImplGenerator {
 
   String _generateUiFactoryConfig(FactoryNames factoryName) {
     return 'final UiFactoryConfig<${names.implName}> '
-        '${factoryName.configName} = UiFactoryConfig(\n'
+        '${factoryName.privateConfigName} = UiFactoryConfig(\n'
         'propsFactory: PropsFactory(\n'
         'map: (map) => ${names.implName}(map),\n'
         'jsMap: (map) => ${names.jsMapImplName}(map),),\n'
-        'displayName: \'${factoryName.consumerName}\');\n\n';
+        'displayName: \'${factoryName.consumerName}\');\n\n'
+        '@Deprecated(r\'This member is incompatible with Dart >2.9.0. Use `${factoryName.privateConfigName}` instead.\')\n'
```

Comment:
1. Dart >**=**2.9.0
2. "Incompatible" is too harsh; it will still work, but the dev experience is worse

There's a lot of context that's hard to include in this string, so perhaps we should just link to it?

```suggestion
        '@Deprecated(r\'Use the private variable, `${factoryName.privateConfigName}`, instead. For information on why this is deprecated, see https://github.com/Workiva/over_react/pull/650\')\n'
```

Reply:
Oh, and maybe we should add a reminder to set the lower bound to the over_react version we expect this to be released in, so that people don't start using these new members and then get pulled into a project with a lower over_react, causing build errors

Reply:
Good idea! I'll update that